dnf-behave-tests/features/steps/cmd.py

```diff
@@ -174,6 +174,11 @@ def step_impl(context):
     context.dnf._set("disable_plugins", False)
 
 
+@behave.given("I use hosts protected packages settings")
+def step_impl(context):
+    context.dnf.setopts.pop("protected_packages", None)
```

Comment:
Why are you removing `protected_packages` key from setopts?

Reply:
Because if we want to use hosts protected packages we cannot override `protected_packages` option using setopt in any way.

If we set it to empty string it doesn't protect any package overriding hosts setting (which this commit does by default in the tests).

Reply:
I see, OK. But this should be eventually moved to generated dnf.conf file or something.

Reply:
Instead of using host's protected packages, you should instead set `yum` as a protected package in the specific test that needs it.

It can be done through the generated config (as well as resetting the protected package list) and is imo a better solution - I've got the generated config implemented in a branch and am just waiting for https://github.com/rpm-software-management/ci-dnf-stack/pull/816 to be merged before submitting a PR.

Reply:
ok, we can move it to the config file, but I think that if we set `yum` as protected manually it will defeat the purpose of the test. This [specific test](https://github.com/rpm-software-management/ci-dnf-stack/blob/master/dnf-behave-tests/features/remove-protected.feature) doesn't test that the protected functionality works, but that dnf when installed protects `yum` (meaning [this](https://github.com/rpm-software-management/dnf/blob/master/etc/dnf/protected.d/yum.conf) exists).  

Now that I think about it, it should rather be `@no_installroot` test. :)

Reply:
I'm taking the PR 816 review.

Reply:
@kontura I see, then it's fine I guess... though I'm not sure we need such a test, you might as well check the contents of that config file as installed from the package is "yum"... :slightly_smiling_face: